sdk/core/azure-core/src/main/java/com/azure/core/util/DateTimeRfc1123.java

```diff
@@ -177,57 +178,174 @@ public static String toRfc1123String(OffsetDateTime dateTime) {
         // ensure datetime is UTC offset.
         dateTime = dateTime.withOffsetSameInstant(ZoneOffset.UTC);
 
-        StringBuilder sb = new StringBuilder(32);
-
+        byte[] bytes = new byte[29];
         final DayOfWeek dayOfWeek = dateTime.getDayOfWeek();
         switch (dayOfWeek) {
-            case MONDAY: sb.append("Mon, "); break;
-            case TUESDAY: sb.append("Tue, "); break;
-            case WEDNESDAY: sb.append("Wed, "); break;
-            case THURSDAY: sb.append("Thu, "); break;
-            case FRIDAY: sb.append("Fri, "); break;
-            case SATURDAY: sb.append("Sat, "); break;
-            case SUNDAY: sb.append("Sun, "); break;
+            case MONDAY:
```

Comment:
Is the perf benefit significant to compromise the readability on this?

Reply:
I'll add my testing numbers but making this change had the following improvements:

Java 17
Previous: 95 ns/op
This: 50 ns/op

Java 8
Previous: 190 ns/op
This: 90 ns/op

Effectively a 2x improvement. Also, since RFC1123 is effectively static this won't need to be modified for a long time